abdul-moize/django_ecommerce
# Feat: Products Management API and Admin site integration

This PR adds `Product` model, api and admin site integration to main branch. A Product model consists of `name, price, price_unit(Currency), quantity, description(optional) and created_by` fields. API allows `Users/Customers` to `view all or one` products and `Admins/Content Managers` to `add, update, delete` products.

To Test this PR, follow these steps:
1. Pull this branch and run: `pip install -r requirements.txt`
2. Run: `python manage.py migrate`
3. Create Users and Super User(see https://github.com/abdul-moize/django_ecommerce/pull/1)
4. Run: `python manage.py runserver`
5. Open Postman and follow steps below to test different modes
- To test Add, login as an Admin or Content Manager and set a post request on `127.0.0.1:8000/products/` with all the fields listed below
![image](https://user-images.githubusercontent.com/88365652/133403140-e55c4c3e-13a0-4989-b7ff-edfa97a2ee77.png)
![image](https://user-images.githubusercontent.com/88365652/133403258-7efa7f05-e457-4521-a77e-fa3818ae1062.png)

- To test update, login as `Admin or CM` and send a put request on `127.0.0.1:8000/products/<id>/`
![image](https://user-images.githubusercontent.com/88365652/133403140-e55c4c3e-13a0-4989-b7ff-edfa97a2ee77.png)
![image](https://user-images.githubusercontent.com/88365652/133404021-8c6d151b-a7f2-4ae0-9a98-96ab8c0e514c.png)

-To test delete, login as `Admin or CM` and send a delete request on `127.0.0.1:8000/products/<id>/`
![image](https://user-images.githubusercontent.com/88365652/133403140-e55c4c3e-13a0-4989-b7ff-edfa97a2ee77.png)
![image](https://user-images.githubusercontent.com/88365652/133404429-1d2209c0-01d7-4908-aaa6-df736383a3c6.png)

- To test list, send a get request at `127.0.0.1:8000/products/`
![image](https://user-images.githubusercontent.com/88365652/133404600-e6c17834-41f3-4e21-9109-f4cdde4f3c6e.png)

- To test retrieve,  send a get request at `127.0.0.1:8000/products/<id>/`
![image](https://user-images.githubusercontent.com/88365652/133404779-7f42805f-4911-416a-9586-502f74756c5a.png)

Admin site looks like this:
All products page
![image](https://user-images.githubusercontent.com/88365652/133406065-921cad64-3ef8-412f-af91-7c6ac7416ffd.png)
Change product page
![image](https://user-images.githubusercontent.com/88365652/133406098-6a21439d-4aa1-472f-a394-08597c81d0c5.png)
Add product page
![image](https://user-images.githubusercontent.com/88365652/133406338-383f7c01-7f0f-43a5-80a2-9e03cc538519.png)
